packages/@react-stately/utils/src/useControlledState.ts

```diff
@@ -12,11 +12,9 @@
 
 import {useCallback, useRef, useState} from 'react';
 
-export function useControlledState<T>(
-  value: T,
-  defaultValue: T,
-  onChange: (value: T, ...args: any[]) => void
-): [T, (value: T, ...args: any[]) => void]  {
+export function useControlledState<T, C = T>(value: T, defaultValue: T | undefined, onChange?: (v: C) => void): [T, (value: T, ...args: any[]) => void];
+export function useControlledState<T, C = T>(value: T | undefined, defaultValue: T, onChange?: (v: C) => void): [T, (value: T, ...args: any[]) => void];
+export function useControlledState<T, C = T>(value: T, defaultValue: T, onChange?: (v: C) => void): [T, (value: T, ...args: any[]) => void] {
```

Comment:
If we are removing the extra spread below, we should remove it from the type here as well. Remind me why we are removing it btw?

Reply:
the onChange caller we have, it’s called with a …args
which means (v: C) => void isn’t quite right, but neither is (v: C, ...rest: any[]) => void , it’s too broad
but I’m not really sure why we have that spread, deleting it still passes every test we have?

Reply:
It shouldn't be any[]
But unless I introduce a third generic, I'm not sure how we could type it

Reply:
it would have to be any